src/ArduinoCloudProperty.cpp

```diff
@@ -23,12 +23,7 @@
 #endif
 
 static unsigned long getTimestamp() {
-  #ifdef ARDUINO_ARCH_SAMD
```

Comment:
Why was the content of this function deleted?

Reply:
Now the code is restored.

> Hi @mirkokurt 👋 I have a very hard time understanding the changes done within this PR. Please outline how the CBOR protocol was extended/changed and why.
> 
> Also I've got a **esspecially a big problem** with test code were the comment was changed but the actual byte array was not changed ([here](https://github.com/arduino-libraries/ArduinoCloudThing/pull/48/commits/82e78d1678ee6e3145eaca570db417116938217d)) with a commit message saying `Fix unit tests`. Why were those tests fixed? I'm looking at magic numbers without explanation in the test code. **This is unacceptable**.
> 
> When writing any code please keep in mind that code is written for your fellow programmers first and for the machine second.

Hi @lxrobotics, I added many comments in the code to help programmers understanding how the library encode and decode the cbor messages when the option of using "light payloads" has been activated. I also updated the PR description to better explain the purpose of the implementation.

Regarding the commit `Fix unit tests`, there are two kind of fixes in the commit:
the first kind is a logic fix:
eg:
-thing.addPropertyReal(color_test, "test", Permission::ReadWrite);
+thing.addPropertyReal(color_test, "test", Permission::ReadWrite, 1);

the addPropertyReal method was invoked without an identifier assigned to the property. So the test was not well written.

The second kind is a fix in the comment that was previously wrong in respect of the byte array.
eg:
-/* [{0: "test", 4: false}] = 81 A2 00 01 04 F4 */
+/* [{0: 1, 4: false}] = 81 A2 00 01 04 F4 */

Reply:
> Why was the content of this function deleted?

By mistake. Now the code is restored.